sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/ApproximateDistinct.java

```diff
@@ -0,0 +1,600 @@
+ * <ul>
+ *    <li> Input Coder : The coder of the objects to combine.
+ *    <li> Precision : Controls the accuracy of the estimation. In general one can expect a
+ *    relative error of about {@code 1.1 / sqrt(2^p)}.
+ *    <li> Sparse Precision : Used to create a sparse representation in order to optimize memory
+ *    and improve accuracy at small cardinalities ({@code < 12000}).
+ * </ul>
+ *
+ * <h2>Using the Transforms</h2>
+ *
+ * <p>The {@link PTransform}s return a {@code PCollection<Long>} corresponding to the estimate
+ * number of distinct elements in the input {@link PCollection} of objects or for each key in a
+ * {@link PCollection} of {@link KV}s.
+ *
+ * <p>By default the Input Coder is inferred at runtime. One should specify a coder only if
+ * custom type is used.
+ * <br>By default the precision is set to {@code 12} for a relative error of around {@code 2%}.
+ * <br>By default the sparse representation is not used. One should use it if the number of
+ * distinct elements in the stream may be less than {@code 12000}.
+ *
+ * <h4>Example 1 : globally default use</h4>
+ * <pre>{@code
+ * PCollection<Integer> input = ...;
+ * PCollection<Long> hllSketch = input.apply(ApproximateDistinct.<Integer>globally());
+ * }</pre>
+ *
+ * <h4>Example 2 : per key default use</h4>
+ * <pre>{@code
+ * PCollection<Integer, String> input = ...;
+ * PCollection<Integer, Long> hllSketches = input.apply(ApproximateDistinct
+ *                .<Integer, String>perKey());
+ * }</pre>
+ *
+ * <h4>Example 3 : tune precision and use sparse representation</h4>
+ * <pre>{@code
+ * int precision = 15;
+ * int sparsePrecision = 25;
+ * PCollection<Double> input = ...;
+ * PCollection<Long> hllSketch = input.apply(ApproximateDistinct
+ *                .<Double>globally()
+ *                .withPrecision(precision)
+ *                .withSparsePrecision(sparsePrecision));
+ * }</pre>
+ *
+ * <h4>Example 4 : use a custom object and tune precision</h4>
+ * <pre>{@code
+ * int precision = 18;
+ * PCollection<MyClass> input = ...;
+ * PCollection<Long> hllSketch = input.apply(ApproximateDistinct.<MyClass>globally()
+ *                  .withInputCoder(MyClassCoder.of())
+ *                  .withPrecision(precision));
+ * }</pre>
+ *
+ * <p>The tuning works exactly the same with {@link #perKey()}.
+ *
+ * <h2>Using the CombineFn</h2>
+ *
+ * <p>The CombineFn outputs the {@link HyperLogLogPlus} structure used to combine the elements.
+ * <br>Nonetheless, it is still possible to retrieve easily the cardinality as a Long in a
+ * {@link PCollection} of {@code HyperLogLogPlus} or {@code KV<K, HyperLogLogPlus>} by using
+ * the {@link RetrieveCardinality} utility class (see example 3).
+ *
+ * <p>By default, one must always specify a coder, using the
+ * {@link ApproximateDistinctFn#create(Coder)} method.
+ * <br>By default, the precision is set to {@code 12} for a relative error of around {@code 2%}.
```

Comment:
Many of these points repeat the documentation above, which is I think also redundant. I'd suggest to say that the CombineFn does the same thing as the transform but it can be used in (some contexts where the transform can't be used), for example (give 1 example of such usage, where the transform can't be used but the CombineFn saves the day)